generator/dart/generated/google_cloud_protobuf/lib/src/protobuf.p.dart

```diff
@@ -14,21 +14,106 @@
 
 part of '../protobuf.dart';
 
-class FieldMaskHelper {
-  /// Encode the field mask as a single comma-separated string.
-  static String encode(FieldMask fieldMask) {
-    return fieldMask.paths?.join(',') ?? '';
+/// `Any` contains an arbitrary serialized message along with a URL that
+/// describes the type of the serialized message.
+class Any extends Message {
+  static const Set<String> _customEncodedTypes = {
+    'google.protobuf.Duration',
+    'google.protobuf.FieldMask'
+  };
+
+  /// The raw JSON encoding of the underlying value.
+  final Map<String, dynamic> json;
+
+  Any({Map<String, dynamic>? json}) : this.json = json ?? {};
+
+  /// Create an [Any] from an existing [message].
+  ///
+  /// [fullyQualifiedName] should be the fully qualified type name of the given
+  /// message (for example, `google.protobuf.Duration`).
+  Any.from(Message message, String fullyQualifiedName) : json = {} {
+    packInto(message, fullyQualifiedName);
   }
 
-  /// Decode the field mask from a single comma-separated string.
-  static FieldMask decode(String format) {
-    return FieldMask(paths: format.split(','));
+  factory Any.fromJson(Map<String, dynamic> json) {
+    return Any(json: json);
+  }
+
+  /// '@type' will be something like
+  /// `type.googleapis.com/google.protobuf.Duration`, or
+  /// `type.googleapis.com/google.rpc.ErrorInfo`.
+  String get _type => json['@type'];
+
+  /// Return the fully qualified name of the contained type.
+  ///
+  /// For example, `google.protobuf.Duration`, or `google.rpc.ErrorInfo`.
+  String get typeName {
+    final type = _type;
+
+    final index = type.lastIndexOf('/');
+    if (index == -1) return type;
+
+    return type.substring(index + 1);
   }
+
+  /// Returns whether the type represented by this `Any` is the same as [name].
+  ///
+  /// [name] should be a fully qualified type name, for example,
+  /// `google.protobuf.Duration` or `google.rpc.ErrorInfo`.
+  bool isType(String name) => typeName == name;
```

Comment:
I think we should verify the prefix is `type.googleapis.com/`.  If we receive an Any with a different prefix something is really wrong and we should not be trying to extract that value.